File: src/LogParser.java
public class LogParser {

    public static LogEntry parseLogLine(String logLine) {

        // Split log line by space
        String[] parts = logLine.split(" ", 5);

        // Extract fields
        String timestamp = parts[0] + " " + parts[1];
        String level = parts[2];
        String source = parts[3];
        String message = parts[4];

        return new LogEntry(timestamp, level, source, message);
    }
}

File: deployy/LogParser.js
class LogEntry {
  constructor(timestamp, level, source, message) {
    this.timestamp = timestamp;
    this.level = level;
    this.source = source;
    this.message = message;
  }

  toString() {
    return `[${this.timestamp}] [${this.level}] [${this.source}] ${this.message}`;
  }
}

module.exports = LogEntry;
